Source/CBLCookieStorage.m

```diff
@@ -0,0 +1,424 @@
+//
+//  CBLCookieStorage.m
+//  CouchbaseLite
+//
+//  Created by Pasin Suriyentrakorn on 2/18/15.
+//  Copyright (c) 2015 Couchbase, Inc. All rights reserved.
+//
+
+#import "CBLCookieStorage.h"
+#import "CBLDatabase.h"
+#import "CBLDatabase+Internal.h"
+#import "CBLMisc.h"
+#import "Logging.h"
+
+
+NSString* const CBLCookieStorageCookiesChangedNotification = @"CookieStorageCookiesChanged";
+NSString* const CBLCookieStorageAcceptPolicyChangedNotification = @"CookieStorageAcceptPolicyChanged";
+
+#define kLocalDocKeyPrefix @"cbl_cookie_storage"
+#define kLocalDocCookiesKey @"cookies"
+
+@interface CBLCookieStorage ()
+- (NSString*) localDocKey;
```

Comment:
FYI, this isn't necessary anymore. The Obj-C compiler doesn't require forward declarations of methods.